sb_vision/tokens.py

```diff
@@ -1,176 +1,34 @@
-
-    if calibration['resolution'] != image_size:
-        raise ValueError(
-            "Model {model} is calibrated for resolution {res_model}, not "
-            "{res_this}".format(
-                model=name,
-                res_model=calibration['resolution'],
-                res_this=image_size,
-            ),
-        )
-
-    if 'marker_size' not in calibration:
-        raise ValueError("Calibrations must have a marker_size")
-
-    return calibration
-
-
-def homography_matrix_to_distance_model_input_vector(homography_matrix):
-    """Convert a 3x3 homography matrix to a vector for distance models."""
-    flattened_vector = homography_matrix.ravel()
-    flattened_vector_as_row_matrix = np.array([flattened_vector])
-
-    quadratic_features = \
-        flattened_vector_as_row_matrix.T.dot(flattened_vector_as_row_matrix)
-    quadratic_features = quadratic_features.ravel()
-
-    all_features = np.hstack((
-        quadratic_features,
-        flattened_vector,
-    ))
-
-    return all_features
-
-
-def _apply_distance_model_component_to_homography_matrix(
-    model_component: Mapping[str, Any],
-    homography_matrix,
-) -> np.float64:
-    flattened_homography_matrix = \
-        homography_matrix_to_distance_model_input_vector(homography_matrix)
-
-    biases = model_component['biases']  # type: np.ndarray
-    intercept = model_component['intercept']  # type: np.float64
-    coefs = model_component['coefs']  # type: np.ndarray
-
-    return (
-        (biases + flattened_homography_matrix).dot(coefs) +
-        intercept
-    )
-
-
-def _get_cartesian(
-    homography_matrix,
-    image_size: Tuple[int, int],
-    distance_model: str,
-    marker_size: Optional[Tuple[float, float]],
-) -> Cartesian:
-    calibration = _get_distance_model(distance_model, image_size)
-    calibrated_marker_size = calibration['marker_size']
-
-    if marker_size is None:
-        size_ratio = 1
-    else:
-        if marker_size[0] != marker_size[1]:
-            raise ValueError("Non-square markers are not supported")
-        # The ratio between the calibrated size and the actual size
-        size_ratio = marker_size[0] / calibrated_marker_size
-
-    in_x = _apply_distance_model_component_to_homography_matrix(
-        calibration['x_model'],
-        homography_matrix,
-    )  # type: np.float64
-    y = 0.0
-    in_z = _apply_distance_model_component_to_homography_matrix(
-        calibration['z_model'],
-        homography_matrix,
-    )  # type: np.float64
-
-    x = in_x * size_ratio
-    z = in_z * size_ratio
-
-    return Cartesian(x, y, z)
-
-
 class Token:
     """Representation of the detection of one token."""
 
-    def __init__(self, id: int, certainty: float=0.0) -> None:
+    def __init__(self,
+                 id: int,
+                 certainty=0,
+                 pixel_coords: Iterable[Tuple[float, float]] = (),
+                 pixel_centre: Tuple[float, float] = (0, 0)) -> None:
```

Comment:
Please can we make the wrapping here consistent with the rest of the repo's multi-line signatures? (see `from_apriltag_detection` for an example)